src/useAuth.ts

```diff
@@ -4,5 +4,5 @@ import { AuthContextProps } from './AuthContextInterface';
 import { AuthContext } from './AuthContext';
 
 export const useAuth = () => {
-  return useContext<AuthContextProps|null>(AuthContext);
+  return useContext<AuthContextProps>(AuthContext);
 };
```

Comment:
Maybe you could add a check if context is truthy otherwise thrown an error?

`export const useAuth = (): AuthContextProps => {
    const context = useContext<AuthContextProps>(AuthContext);

    if (!context) {
        throw new Error('useAuth should only be used in AuthContext');
    }

    return context;
};`

Reply:
agreed and fixed